hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/AbstractGenericRecordBuilder.java

```diff
@@ -93,6 +93,10 @@ public GenericRecordBuilder setInt16(@Nonnull String fieldName, short value) {
     @Nonnull
     @Override
     public GenericRecordBuilder setGenericRecord(@Nonnull String fieldName, @Nullable GenericRecord value) {
+        if (value != null && !(value instanceof CompactGenericRecord)) {
```

Comment:
Minor: Since there are two implementations (and it is possibly going to be like this for a very long time), we can do `value instanceof PortableGenericRecord` instead

Reply:
Yes but I think this one is better in terms of readability and logic(if it is not compact in compact builder we throw) and very long term in which we may add a new one.

Reply:
OK, that is fine as well 👍 BTW, what do you think about putting this check into a function (as it is duplicated in two places) and calling it here and below?

Reply:
I am fine with that. Doing..

Reply:
Done. I also noticed I forgot a null check in portable genrecord builder. With the extracted method, it is fixed.